src/onig-scanner.cc

```diff
@@ -58,8 +58,8 @@ OnigScanner::OnigScanner(Local<Array> sources) {
   regExps.resize(length);
 
   for (int i = 0; i < length; i++) {
-    String::Utf8Value utf8Value(sources->Get(i));
-    regExps[i] = shared_ptr<OnigRegExp>(new OnigRegExp(string(*utf8Value)));
+    Local<String> source = Local<String>::Cast(sources->Get(i));
+    regExps[i] = shared_ptr<OnigRegExp>(new OnigRegExp(OnigString(source)));
```

Comment:
Drop the type aliases in `onig-searcher.h`, and use a fully qualified reference to `std::shared_ptr<T>` as you've done elsewhere? This file probably never should have relied on those aliases in that header.

PS. Dropping those aliases will obviously require updating the type references in other files, most notably `onig-searcher.cc`.